### PR TITLE
added typings

### PR DESCRIPTION
**Pull request by donaldpipowitch**
Thank you very much for glamor. I really like it and use it together with TypeScript, so I added some basic typings.

The core lib is declared in `index.d.ts` and is based on the API documentation. 

All other files (e.g. for `glamor/react`) have only _very_ basic typings (they only declare the keys of exported variables, but no specific types). They are a good starting point for others to make them better. I only use the core lib and maybe something like `glamor/react` could be moved into its own repo in the future...? 

Maybe I can convince you to switch completely to TypeScript? :) Here is a nice article about this topic: http://staltz.com/all-js-libraries-should-be-authored-in-typescript.html
It is great for library authors and consumers.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,301 @@
+export interface CSSProperties {
+  /**
+   * In dev mode, adding a `label` string prop will reflect its value in devtools. Useful
+   * when debugging, and a good alternative to 'semantic' classnames.
+   */
+  label?: string;
+  // for now just allow everything
+  [propertyName: string]: any;
```
Comment (donaldpipowitch) [inline index.d.ts]: I hope we can re-use some standard interface in the future here. See https://github.com/Microsoft/TypeScript/issues/10820.

**Comment by threepointone** [comment]
This is awesome! I'll admit I have no experience with typescript, so I have some questions. 

- how are these consumed? like, in your project, how do you import these typings?
- should I be including these files in the npm package as well? ie - should I modify the `files` field as well?
- is it possible to move these into a `typings` folder? Or does it have to be at root?
- I actually *like* that `/react` is in the same package, it doesn't make download slower or anything; does it matter/bother you much?

I do like typescript (and flow, inference is neat), just haven't gotten around to trying it out. do you have any links/resources for me to read? swamped now with another side project, but I'd love to get to it in a week or so. 

Thanks so so much for this PR, this is going to be great!

**Comment by donaldpipowitch** [comment]
> how are these consumed? like, in your project, how do you import these typings?

If you import `glamor` TypeScript checks the declarations set in the `typings` field [here](https://github.com/threepointone/glamor/pull/49/commits/a9f4a65f8c62e8801b03966319ddfaa4251087ff#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R6). It is basically like a `main` field for declarations :)

If you import `glamor/react` and similar files TypeScript looks for `.d.ts` files with the name `react`. (It just appends `.d.ts` to your import path internally.)

> should I be including these files in the npm package as well? ie - should I modify the files field as well?

Oh yeah, sorry. It must be in the npm package. I should add them.

> is it possible to move these into a typings folder? Or does it have to be at root?

Normally you can move all declarations in a `typings/` folder. TypeScript just looks for the file given in the `typings` field. But I'm not sure, if we can do it here. `glamor/react` and others aren't re-exported by the main file. They are basically standalone modules - everything acts as its own main. (As far as I can see it.) I don't know if it possible to declare type declarations for other files here. It _could_ be possible by using `declare module 'glamor/react'` in the `index.d.ts`, but I haven't tried it.

> I actually like that /react is in the same package, it doesn't make download slower or anything; does it matter/bother you much?

Currently it doesn't matter much for me. But as I said it looks `glamor/react` is treated as its "own module". This can become harry for  every tool which expects _a single entry point_. The `typings` field shows this. I guess tools which use something like `esnext:main` would have similar problems, if you would use this, because you can't have multiple mains.

> do you have any links/resources for me to read?

Mainly http://staltz.com/all-js-libraries-should-be-authored-in-typescript.html as an introduction. Check the official blogs from Flow and TypeScript to get a feeling for the actual progress and read a little bit of their docs. Lastly... try VS Code with a TypeScript project with good documentation. VS Code shows _great_ inline documentation. It makes autocompletion and refactoring very easy.

**Comment by threepointone** [comment]
alright, I'll merge these in for now and we'll see how they work out. We'll take a call on polishing these as we get feedback. Thank you!

**Comment by threepointone** [comment]
This is now live in 2.15.4. huzzah!

**Pull request merged**